### ISSUE TITLE
Outer product of two vectors gives wrong result

### ISSUE THREAD

**Issue by kostrykin** [flatironinstitute/sparse_dot#4]
Using `dot_product_mkl` to compute the outer product of two matrices gives the wrong result. By definition of the outer product, we would expect obtaining a matrix, however, we get a vector.

Lets say I have two matrices, `A1` and `A2`, where both have only a single column:
```python
n = 1
A1 = np.random.randn(6, n)
A2 = np.random.randn(10, n)
A2[A2 < 0] = 0
A2 = scipy.sparse.csr_matrix(A2)
```

Checking `dot_product_mkl(A1, A2.T).shape` gives `(1, 10)` which is not correct. In accordance to `(A1 @ A2.T).shape` we would expect a `(6, 10)`-matrix. Is this a bug or is this behavior on purpose? Note that the correct result is obtained if `n=2` is used instead.

**Comment by asistradition** [comment]
Yeah that's definitely a bug (and very much my fault). 

There was a logic error that was calling matrix (dot) vector instead of the correct matrix (dot) matrix which produces the right outer product. Should be fixed in 0434e0ae86dd4171c1e300ce6c32b201f7cb0405, and I'll try to get it packaged at some point today.

**Comment by asistradition** [comment]
Packaged and released this fix (and also the code that calls to `sparse_syrk`) as v0.5.1
